Cogs/settings.py

```diff
@@ -1,15 +1,19 @@
 import discord
 from discord.ext import commands
 
+from Cogs.Helpers.helpers import checkTextChannel
 from Cogs.db import dbconnect
 
 class Settings(commands.Cog):
     def __init__(self, bot):
         self.bot = bot
 ###########################################################################
-    @commands.command(name = "setwelcomechannel") #Sets the channel that member-join messages are sent
+    @discord.slash_command(name = "setwelcomechannel", debug_guilds=[887366492730036276]) #Sets the channel that member-join messages are sent
     @commands.has_permissions(manage_guild = True)
-    async def setwelcomechannel(self, ctx, channel: discord.TextChannel):
+    async def setwelcomechannel(self, ctx, channel: discord.Option(discord.SlashCommandOptionType.channel, "Select channel")):
+        if (not checkTextChannel(channel)):

```

Comment:
Could turn this check into an annotation on the method for readability

Reply:
Tried this, didn't work because using a check changes the type of parameter to `class discord.commands.context.ApplicationContext`, and I tried narrowing that down to a text channel and was unable to. The .channel of that didn't yield any helpful results either unfortunately.